python/cudf/cudf/io/parquet.py

```diff
@@ -64,11 +64,23 @@ def read_parquet(
 
 
 @ioutils.doc_to_parquet()
-def to_parquet(df, path, *args, **kwargs):
+def to_parquet(
+    df,
+    path,
+    compression=None,
+    statistics="ROWGROUP",
+    engine="cudf",
+    index=None,
+    partition_cols=None,
+    *args,
+    **kwargs,
+):
     """{docstring}"""
-    warnings.warn(
-        "Using CPU via PyArrow to write Parquet dataset, this will "
-        "be GPU accelerated in the future"
-    )
-    pa_table = df.to_arrow()
-    pq.write_to_dataset(pa_table, path, *args, **kwargs)
+
+    if engine == "cudf":
+        return libcudf.parquet.write_parquet(
+            df, path, compression=compression, statistics=statistics
+        )
+    else:
+        pa_table = df.to_arrow()
+        pq.write_to_dataset(pa_table, path, *args, **kwargs)
```

Comment:
Need to pass `index` and `partition_cols` into here explicitly now since they're no longer packed in `kwargs`

Reply:
Added "partition_cols" but didn't see "index" at https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_to_dataset.html? Am I missing something?

Reply:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html